wandb/cli/cli.py

```diff
@@ -484,6 +491,8 @@ def sync(
     clean_old_hours=24,
     clean_force=None,
 ):
+    # TODO: rather unfortunate, needed to avoid creating a `wandb` directory
+    os.environ["WANDB_DIR"] = TMPDIR.name
```

Comment:
@raubitsj this one really sucks.  I guess we can address it once we get rid of wandb.old and `_get_cling_api`... Not sure if it was worth it to dig deeper.

Reply:
Other people took a pass at not creating wandb dirs, but i think it wasnt done cleanly so it keeps coming back